zio-http/src/main/scala/zhttp/http/Http.scala

```diff
@@ -168,6 +168,62 @@ sealed trait Http[-R, +E, -A, +B] { self =>
   def unwrap[R1 <: R, E1 >: E, C](implicit ev: B <:< ZIO[R1, E1, C]): Http[R1, E1, A, C] =
     self.flatMap(Http.fromEffect(_))
 
+  /**
+   * Returns an Http that peeks at the success of this Http.
+   */
+  def tap[R1 <: R, E1 >: E, A1 <: A](f: B => Http[R1, E1, Any, Any]): Http[R1, E1, A, B] =
+    self.flatMap(v => f(v).as(v))
+
+  /**
+   * Returns an Http that effectfully peeks at the success of this Http.
+   */
+  def tapM[R1 <: R, E1 >: E](f: B => ZIO[R1, E1, Any]): Http[R1, E1, A, B] =
+    self.tap(v => Http.fromEffect(f(v)))
+
+  /**
+   * Returns an Http that peeks at the failure of this Http.
+   */
+  def tapError[R1 <: R, E1 >: E](f: E => Http[R1, E1, Any, Any]): Http[R1, E1, A, B] =
+    self.foldM(
+      e => f(e) *> Http.fail(e),
+      Http.succeed,
+      Http.empty,
+    )
+
+  /**
+   * Returns an Http that effectfully peeks at the failure of this Http.
+   */
+  def tapErrorM[R1 <: R, E1 >: E](f: E => ZIO[R1, E1, Any]): Http[R1, E1, A, B] =
+    self.tapError(e => Http.fromEffect(f(e)))
+
+  /**
+   * Returns an Http that peeks at the success, failed or empty value of this Http.
+   */
+  def tapAll[R1 <: R, E1 >: E](
+    f: E => Http[R1, E1, Any, Any],
+    g: B => Http[R1, E1, Any, Any],
+    h: => Http[R1, E1, Any, Any],
```

Comment:
`foldM` will immediately evaluate `h`. I am not sure passing by name adds much value. There is bit of a performance penalty also associated with it.



```suggestion
    h: Http[R1, E1, Any, Any],
```